### PR TITLE
Fix up gke-deploy docs.

### PR DESCRIPTION
**Pull request by joonlim**

### CODE REVIEW

**Comment by joonlim** [comment]
@ImJasonH Docs were messed up due to an auto-formatter.

**Comment by ImJasonH** [comment]
Are these tables generated by cobra? If not, can they be? It'd be nice to have one canonical documentation in Go which gets generated into Markdown.

**Comment by joonlim** [comment]
> Are these tables generated by cobra? If not, can they be? It'd be nice to have one canonical documentation in Go which gets generated into Markdown.

Is there a cobra command that generates tables?

**Comment by ImJasonH** [comment]
> > Are these tables generated by cobra? If not, can they be? It'd be nice to have one canonical documentation in Go which gets generated into Markdown.
> 
> Is there a cobra command that generates tables?

Not tables, as far as I know, but it can generate Markdown files at least. See [`crane` docs](https://github.com/google/go-containerregistry/blob/master/cmd/crane/doc/crane.md), generated from [here](https://github.com/google/go-containerregistry/blob/master/cmd/crane/help/main.go). We ensure the generated docs are up-to-date [in CI](https://github.com/google/go-containerregistry/blob/master/.travis.yml#L27-L28).

**Comment by joonlim** [comment]
> > > Are these tables generated by cobra? If not, can they be? It'd be nice to have one canonical documentation in Go which gets generated into Markdown.
> > 
> > 
> > Is there a cobra command that generates tables?
> 
> Not tables, as far as I know, but it can generate Markdown files at least. See [`crane` docs](https://github.com/google/go-containerregistry/blob/master/cmd/crane/doc/crane.md), generated from [here](https://github.com/google/go-containerregistry/blob/master/cmd/crane/help/main.go). We ensure the generated docs are up-to-date [in CI](https://github.com/google/go-containerregistry/blob/master/.travis.yml#L27-L28).

I copied the structure in the example you gave. For some reason, I can't use the github.com/spf13/cobra/doc library as long as the vendor dir exists.

**Comment by joonlim** [comment]
```
Step #0: ?   	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd	[no test files]
Step #0: ?   	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd/apply	[no test files]
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd/common	0.010s
Step #0: ?   	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd/prepare	[no test files]
Step #0: ?   	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd/run	[no test files]
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/core/cluster	0.006s
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/core/gcp	0.003s
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/core/image	0.003s
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/core/resource	0.061s
Step #0: ok  	github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/deployer	35.033s
Step #0: # github.com/cpuguy83/go-md2man/md2man
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:11:16: undefined: blackfriday.EXTENSION_NO_INTRA_EMPHASIS
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:12:16: undefined: blackfriday.EXTENSION_TABLES
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:13:16: undefined: blackfriday.EXTENSION_FENCED_CODE
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:14:16: undefined: blackfriday.EXTENSION_AUTOLINK
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:15:16: undefined: blackfriday.EXTENSION_SPACE_HEADERS
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:16:16: undefined: blackfriday.EXTENSION_FOOTNOTES
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:17:16: undefined: blackfriday.EXTENSION_TITLEBLOCK
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:19:29: too many arguments to conversion to blackfriday.Markdown: blackfriday.Markdown(doc, renderer, extensions)
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:19:9: cannot use &roffRenderer literal (type *roffRenderer) as type blackfriday.Renderer in return argument:
Step #0: 	*roffRenderer does not implement blackfriday.Renderer (missing RenderFooter method)
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: undefined: blackfriday.LIST_TYPE_ORDERED
Step #0: /go/pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: too many errors
Finished Step #0
```

**Comment by joonlim** [comment]
Here's the problem I'm facing:

If I add `_ "github.com/spf13/cobra/doc"` to any of my source files and run `./update-dep.sh`, I see this:

```
➜  ./update-deps.sh  
go: finding github.com/BurntSushi/xgb latest
go: finding github.com/coreos/pkg latest
go: finding sigs.k8s.io/structured-merge-diff latest
go: finding github.com/google/go-containerregistry latest
go: finding github.com/elazarl/goproxy latest
go: finding google.golang.org/genproto latest
go: finding gopkg.in/tomb.v1 latest
go: finding github.com/armon/consul-api latest
go: finding github.com/PuerkitoBio/urlesc latest
go: finding golang.org/x/sync latest
go: finding golang.org/x/sys latest
go: finding github.com/xiang90/probing latest
go: finding golang.org/x/oauth2 latest
go: finding github.com/google/pprof latest
go: finding golang.org/x/time latest
go: finding golang.org/x/exp latest
go: finding golang.org/x/tools latest
go: finding github.com/docker/spdystream latest
go: finding github.com/mwitkow/go-conntrack latest
go: finding golang.org/x/net latest
go: finding golang.org/x/lint latest
go: finding gopkg.in/check.v1 latest
go: finding github.com/mailru/easyjson latest
go: finding github.com/tmc/grpc-websocket-proxy latest
go: finding golang.org/x/crypto latest
go: finding k8s.io/gengo latest
go: finding k8s.io/apimachinery latest
go: finding k8s.io/kube-openapi latest
go: finding github.com/coreos/go-systemd latest
go: finding github.com/mxk/go-flowrate latest
go: finding golang.org/x/mobile latest
go: finding github.com/golang/groupcache latest
go: finding github.com/munnerz/goautoneg latest
go: finding golang.org/x/image latest
go: finding github.com/nightlyone/lockfile latest
go: finding github.com/kr/logfmt latest
go: finding github.com/prometheus/client_model latest
go: finding github.com/dgryski/go-sip13 latest
go: finding github.com/modern-go/concurrent latest
go: finding github.com/jstemmer/go-junit-report latest
go: finding github.com/alecthomas/template latest
go: finding github.com/golang/glog latest
go: finding github.com/alecthomas/units latest
go: finding github.com/sdboyer/constext latest
# github.com/cpuguy83/go-md2man/md2man
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:11:16: undefined: blackfriday.EXTENSION_NO_INTRA_EMPHASIS
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:12:16: undefined: blackfriday.EXTENSION_TABLES
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:13:16: undefined: blackfriday.EXTENSION_FENCED_CODE
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:14:16: undefined: blackfriday.EXTENSION_AUTOLINK
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:15:16: undefined: blackfriday.EXTENSION_SPACE_HEADERS
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:16:16: undefined: blackfriday.EXTENSION_FOOTNOTES
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:17:16: undefined: blackfriday.EXTENSION_TITLEBLOCK
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:19:29: too many arguments to conversion to blackfriday.Markdown: blackfriday.Markdown(doc, renderer, extensions)
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:19:9: cannot use &roffRenderer literal (type *roffRenderer) as type blackfriday.Renderer in return argument:
	*roffRenderer does not implement blackfriday.Renderer (missing RenderFooter method)
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: undefined: blackfriday.LIST_TYPE_ORDERED
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: too many errors
```

**Comment by joonlim** [comment]
Found a mention of the problem here: https://github.com/spf13/cobra/issues/805

I'm not sure yet how to fix this in the vendor

**Comment by joonlim** [comment]
Looks to me that github.com/spf13/cobra/doc is importing an older version of github.com/cpuguy83/go-md2man. Looking into this.

**Comment by ImJasonH** [comment]
Thanks for looking into this. In the meantime if you just want to get this PR merged and work on generating docs in a future change that's fine too, just let me know.

**Comment by joonlim** [comment]
I believe this is the issue:

`cobra/doc` relies on `https://github.com/cpuguy83/go-md2man/`, which relies on `https://github.com/russross/blackfriday/`.

The latest release of `https://github.com/cpuguy83/go-md2man/` is [1.0.10](https://github.com/cpuguy83/go-md2man/releases), which is broken because it calls some variables in ` blackfriday` that do not exist in the version that `go-md2man` is relying on: [v2.0.1](https://github.com/russross/blackfriday/releases).

However, running without vendor works because `https://github.com/cpuguy83/go-md2man`'s master branch doesn't use said missing variables.

**Comment by joonlim** [comment]
^ It seems that [`https://github.com/cpuguy83/go-md2man/ 1.0.10`](https://github.com/cpuguy83/go-md2man/releases/tag/v1.0.10) is supposed to reference blackfriday 1.5.2. I'm looking into why it's not working for me.

**Comment by joonlim** [comment]
^ I think the problem may be in the `gke-deploy` source.

If I have this in a separate:
```
package main

import (
	"fmt"
	"log"
	"os"

	"github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/cmd"
	"github.com/spf13/cobra"
	"github.com/spf13/cobra/doc"
)

var dir string
var root = &cobra.Command{
	Use:   "gendoc",
	Short: "Generate gke-deploy's help docs",
	Args:  cobra.NoArgs,
	Run: func(*cobra.Command, []string) {
		if err := doc.GenMarkdownTree(cmd.NewCommand(), dir); err != nil {
			log.Fatalln(err)
		}
	},
}

func init() {
	root.Flags().StringVarP(&dir, "dir", "d", ".", "Path to directory in which to generate docs")
}

func main() {
	if err := root.Execute(); err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
}
```

and run `go get` on the dir, it fails:

```
➜  go get -u
go: finding github.com/GoogleCloudPlatform/cloud-builders/gke-deploy latest
go: finding github.com/PuerkitoBio/urlesc latest
go: finding github.com/BurntSushi/xgb latest
^Cgo: finding github.com/armon/consul-api latest
go: finding github.com/mxk/go-flowrate latest
go: finding github.com/coreos/go-systemd latest
go: finding google.golang.org/genproto latest
go: finding golang.org/x/net latest
go: finding github.com/mwitkow/go-conntrack latest
go: finding golang.org/x/lint latest
go: finding k8s.io/kube-openapi latest
go: finding github.com/golang/glog latest
go: finding github.com/munnerz/goautoneg latest
go: finding golang.org/x/tools latest
go: finding github.com/xiang90/probing latest
go: finding github.com/modern-go/concurrent latest
go: finding github.com/mailru/easyjson latest
go: finding github.com/docker/spdystream latest
go: finding golang.org/x/crypto latest
go: finding golang.org/x/sys latest
go: finding gopkg.in/check.v1 latest
go: finding github.com/kr/logfmt latest
go: finding github.com/google/go-containerregistry latest
go: finding github.com/elazarl/goproxy latest
go: finding github.com/google/pprof latest
go: finding golang.org/x/sync latest
go: finding golang.org/x/oauth2 latest
go: finding golang.org/x/exp latest
go: finding github.com/tmc/grpc-websocket-proxy latest
go: finding golang.org/x/mobile latest
go: finding k8s.io/apimachinery latest
go: finding gopkg.in/tomb.v1 latest
go: finding sigs.k8s.io/structured-merge-diff latest
go: finding github.com/jstemmer/go-junit-report latest
go: finding github.com/coreos/pkg latest
go: finding github.com/prometheus/client_model latest
go: finding k8s.io/gengo latest
go: finding github.com/golang/groupcache latest
go: finding golang.org/x/time latest
cgo: finding golang.org/x/image latest
ago: finding github.com/alecthomas/units latest
tgo: finding github.com/alecthomas/template latest
 mgo: finding github.com/dgryski/go-sip13 latest
ai# github.com/cpuguy83/go-md2man/md2man
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:11:16: undefined: blackfriday.EXTENSION_NO_INTRA_EMPHASIS
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:12:16: undefined: blackfriday.EXTENSION_TABLES
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:13:16: undefined: blackfriday.EXTENSION_FENCED_CODE
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:14:16: undefined: blackfriday.EXTENSION_AUTOLINK
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:15:16: undefined: blackfriday.EXTENSION_SPACE_HEADERS
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:16:16: undefined: blackfriday.EXTENSION_FOOTNOTES
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:17:16: undefined: blackfriday.EXTENSION_TITLEBLOCK
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/md2man.go:19:29: too many arguments to conversion to blackfriday.Markdown: blackfriday.Markdown(doc, renderer, extensions)
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:19:9: cannot use &roffRenderer literal (type *roffRenderer) as type blackfriday.Renderer in return argument:
	*roffRenderer does not implement blackfriday.Renderer (missing RenderFooter method)
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: undefined: blackfriday.LIST_TYPE_ORDERED
../../../../../pkg/mod/github.com/cpuguy83/go-md2man@v1.0.10/md2man/roff.go:102:11: too many errors
```

**Comment by joonlim** [comment]
After many attempts, I've still not been able to find the root of the solution. One clue is that the `go.sum` file contains both `https://github.com/russross/blackfriday/` v1.5.2 and v2.0.1.

I've decided to punt adding the help tool for now.

I've added the generated docs to this PR.

I've also copied over the work from [@dorbin's PR](https://github.com/GoogleCloudPlatform/cloud-builders/pull/514) to preemptively deal with merge conflicts.

File: gke-deploy/VENDOR-LICENSE [hunk 1]
```diff
@@ -1,4 +1,211 @@
 
+===========================================================
+Import: github.com/GoogleCloudPlatform/cloud-builders/gke-deploy/vendor/github.com/GoogleCloudPlatform/cloud-builders/gke-deploy
```
Comment (ImJasonH) [inline gke-deploy/VENDOR-LICENSE]: This indicates you're vendoring yourself somewhere, which you definitely don't want to be doing.

File: gke-deploy/go.mod [hunk 2]
```diff
@@ -1,58 +1,17 @@
-module github.com/GoogleCloudPlatform/cloud-builders/gke-deploy
+module github.com/GoogleCloudPlatform/cloud-builders
```
Comment (ImJasonH) [inline gke-deploy/go.mod]: This is probably also a bad idea, since it will likely conflict with gcs-fetcher's go.mod in confusing ways.

Comment (joonlim) [reply gke-deploy/go.mod]: Sorry, not sure why this is happening when I run my `update-deps.sh` script

File: gke-deploy/VENDOR-LICENSE (hunk 1 above)
Comment (joonlim) [reply gke-deploy/VENDOR-LICENSE]: This must've started happening because of generate-docs go binary.

File: gke-deploy/go.mod (hunk 2 above)
Comment (joonlim) [reply gke-deploy/go.mod]: ^ Fixed by adding `gke-deploy` to the end of that url. I'm testing gcloud builds submit right now before pushing.

Comment (joonlim) [reply gke-deploy/go.mod]: Fixed

**Pull request merged**